account_invoice_change_currency/models/account_change_currency.py

```diff
@@ -0,0 +1,174 @@
+        help="Set new currency rate to apply on the invoice\n."
+        "This rate will be taken in order to convert amounts between the "
+        "currency on the invoice and last currency",
+    )
+    created_invoice_currency = fields.Many2one(
+        "res.currency",
+        help="""Save the original currency from the company""",
+    )
+    created_invoice_rate = fields.Float(help="""Save previous custom rate""")
+    last_currency = fields.Many2one(
+        "res.currency",
+        help="""Save the original currency from the company""",
+    )
+    old_rate = fields.Float(help="""Save previous custom rate""")
+    is_forced_rate = fields.Boolean(default=False, help="""Save force rate state""")
+    is_first_currency_equal_to_currency_id = fields.Boolean(
+        compute="_compute_is_first_currency_equal_to_currency_id",
+        store=True,
+        help="""Save a boolean value to recognize if current currency
+        is equal to currency when the invoice was created""",
+    )
+
+    @api.model
+    def create(self, values):
+        invoice = super(AccountMove, self).create(values)
+        if self.env.context.get("force_rate"):
+            invoice._toggle_forced_rate()
+        invoice._onchange_currency_change_rate()
+        invoice.write(
+            {
+                "last_currency": invoice.currency_id,
+                "old_rate": invoice.custom_rate,
+                "created_invoice_rate": invoice.custom_rate,
+                "created_invoice_currency": invoice.currency_id,
+            }
+        )
+        return invoice
+
+    def action_account_change_currency(self):
+        precision = self.env["decimal.precision"].precision_get("Payment Terms")
+        today = fields.Date.context_today(self)
+        for invoice in self.filtered(lambda x: x.state == "draft").with_context(
+            check_move_validity=False,
+        ):
+            from_currency = invoice.last_currency
+            if not from_currency:
+                continue
+            invoice_date = invoice.invoice_date or today
+            ctx = {"company_id": invoice.company_id.id, "date": invoice_date}
+            currency = invoice.with_context(**ctx).currency_id
+            currency_skip = invoice.created_invoice_currency
+            old_rate = invoice.old_rate
+            new_rate = invoice.custom_rate
+            rate_skip = currency.with_context(**ctx)._get_conversion_rate(
+                currency_skip, currency, invoice.company_id, invoice_date
+            )
+            if not new_rate:
+                new_rate = rate_skip
+            if (
+                from_currency == currency
+                and new_rate
+                and old_rate
+                and float_compare(new_rate, old_rate, precision_digits=precision) == 0
+            ):
+                if not invoice.custom_rate or old_rate == new_rate:
+                    continue
+                if float_compare(new_rate, old_rate, precision_digits=precision) == 0:
```

Comment:
Can be joined with the above conditional

Reply:
Done!